bot/bpm_gifs/generator/adjust_video.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from ffmpeg import FFmpeg
+from pathlib import Path
+
+from bot.utils.config import tmp_gifs_dir
+
+logger = logging.getLogger(__name__)
+
+
+def adjust_video_tempo(video_path: Path, video_bpm: float, target_bpm: float) -> bytes:
+    current_interval_sec = 60.0 / video_bpm
+    target_interval_sec = 60.0 / target_bpm
+
+    speed_factor = current_interval_sec / target_interval_sec
+
+    output_file = tmp_gifs_dir / f"{video_path.stem}_{int(target_bpm)}.mp4"
+
+    if output_file.exists():
+        logger.info(f"Using cached video: {output_file}")
+        with open(output_file, 'rb') as f:
+            content = f.read()
+
+        return content
+
+    logger.info(f"Adjusting tempo from {video_bpm} -> {output_file}")
+
+    ffmpeg = (
+        FFmpeg()
+        .option("y")
+        .input(video_path)
+        .output(
+            output_file,
+            # there was pipe:1 before but mp4 not support streaming
+            f='mp4',
+            vf=f"setpts=(PTS-STARTPTS)/{speed_factor}",
+            af=f"atempo={speed_factor}",
+            vcodec="libx264",
+            preset="slow",
+            crf=24,
+
+        )
+    )
+
+    try:
+        ffmpeg.execute()
+        with open(output_file, 'rb') as f:
+            content = f.read()
+    except Exception as e:
+        raise e
+    finally:
+        if output_file.exists():
+            output_file.unlink()
```

Comment:
can be replaced with
```python
output_file.unlink(missing_ok=True)
```
https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink